### ISSUE TITLE
The categories in the blog plugin are not displaying completely

### ISSUE THREAD

**Issue by shenweiyan** [squidfunk/mkdocs-material#6390]
### Bug description

As mentioned in https://github.com/squidfunk/mkdocs-material/issues/6388, there's another issue: even when a post has only one category, sometimes it doesn't show up. For example, this article's category of  [2023-11-23-translate.md](https://raw.githubusercontent.com/shenweiyan/mkdocs-blog/main/docs/blog/posts/2023-11-23-translate.md) isn't displayed in https://shenweiyan.github.io/mkdocs-blog/blog/. I've tried multiple times but couldn't figure out where the issue lies. I hope someone can help me troubleshoot this.

### Related links

- https://github.com/shenweiyan/mkdocs-blog
- https://shenweiyan.github.io/mkdocs-blog/blog/

### Reproduction

[9.4.11-mkdocs-blog-category-missing.zip](https://github.com/squidfunk/mkdocs-material/files/13454197/9.4.11-mkdocs-blog-category-missing.zip)

### Steps to reproduce

1. mkdocs serve

### Browser

Chrome

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and providing the excellent reproduction! This is, in fact, a configuration error, because the default slugify function does not support Unicode characters, which slugifies `"资讯"` to `""` (empty category). Changing the configuration fixes the problem:

``` yaml
plugins:
  - blog:
      post_slugify: !!python/object/apply:pymdownx.slugs.slugify
        kwds:
          case: lower
```

__However__, I took this as an opportunity to improve the situation. We want to make it as simple as possible to get started, and using a slugification function that is not Unicode-aware as a default makes configuration more complex and the experience less enjoyable. For this reason, I've changed the default slugification function to the one mentioned above in d43626a83, so starting from this commit, there's no need for configuration.

**Comment by squidfunk** [comment]
The latest commit on Insiders now also changes this for the [tags plugin](https://squidfunk.github.io/mkdocs-material/plugins/tags/) – there should be no more need to specify this setting, except for when you want to customize slugification in non-standard ways.

**Comment by squidfunk** [comment]
Changes of https://github.com/squidfunk/mkdocs-material/commit/d43626a838f8ac7761fccd3eeb214946186f40fa released as part of [9.4.12](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.12).
